engine/lite/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/ScenarioInterpreterFactory.scala

```diff
@@ -362,50 +367,69 @@ object ScenarioInterpreterFactory {
 
     // First we compute scenario parts compiled so far. Then we search for JoinResults and invoke joinPart
     // We know that we'll find all results pointing to join, because we sorted the parts
+    // Where join are placed in parallel sorting can be not enough - see `computePartsToRunAndPass`
```

Comment:
Will such a comment be true? (I am writing as I understand it, so correct me if I am wrong)

> due to `PartSort.sort` done in `ProcessCompilerBase.compileSources` we have guarantee that `join part` will be put in order **after** parts that that references to them (in branches). Still, because of possibility to create nested/parallel unions, we have to invoke currently iterated join **only** for  `part results` that **refer to current join**, and **pass other results further** (they should be catch by joins which were put later due to topological sort).